StRoot/StETofUtil/etofHelperFunctions.cxx

```diff
@@ -0,0 +1,305 @@
+        //return 0.;
+    }
+
+    
+    double tstart = 0.;
+    if( nWestTubes + nEastTubes ) {
+        tstart = tMean;
+
+        if( fabs( pVtxZ ) < 200. ) {
+            LOG_DEBUG << "calculateVpdTstart():  --  Vpd start time before pVtxZ ( " << pVtxZ << " ) correction: " << tstart << endm;            
+
+            tstart -= ( ( nEastTubes - nWestTubes ) * pVtxZ / ( c_light * nanosecond ) ) / ( nWestTubes + nEastTubes );
+        }
+
+    }
+
+
+    if( tstart > bTofClockRange ) {
+        tstart -= bTofClockRange;
+    }
+    else if( tstart < 0. ) {
+        tstart += bTofClockRange;
+    }
+
+    LOG_INFO << "calculateVpdTstart(): -- sum: " << tMean << " nWest: " << nWest << " nEast: " << nEast;
+    LOG_INFO << "  --  Vpd start time: " << tstart << endm;
+
+    return tstart;
+}    
+// *******************************************************************************************************
+
+double
+updateCyclicRunningMean( double mean, double value, int& count, double bTofClockRange )
+{
+    double valIn = value;
+    if( mean - value < -0.9 * bTofClockRange ) {
+        valIn -= bTofClockRange;  
+    } 
+    else if( mean - value > 0.9 * bTofClockRange ) {
+        valIn += bTofClockRange;
+    }
+
+
+    double scaling = 1. / ( count + 1. );
+
+    double valOut  = valIn * scaling + mean * ( 1. - scaling );
+    count++;
+
+    //LOG_INFO << "old mean: " << mean << "  scaling: " << scaling << " value in: " << valIn << "  new mean: " << valOut << endm;
+    
+    return valOut;
+}
+
+
+
+// *******************************************************************************************************
+// calculate pathlength of a helix between two points 
+double
+tofPathLength( const StThreeVectorD&  beginPoint, const StThreeVectorD& endPoint, const double& curvature )
+{
+
+  StThreeVectorD vdif = endPoint - beginPoint;
+  double C = vdif.perp();  
+  double s_perp = C;
+  if ( curvature ) {
+    double R = 1 / curvature;
+    s_perp = 2 * R * asin( C / ( 2 * R ) );
+  }
+
+  double s_z = fabs( endPoint.z() - beginPoint.z() );
+  double value = sqrt( s_perp * s_perp + s_z * s_z );
+
+  return value;
+}
+// *******************************************************************************************************
+// calculate expected time of flight of particle hypothesis
+double
+expectedTimeOfFlight( const double& pathLength, const double& momentum, const double& mass )
+{
+    double inverseBeta = sqrt( 1 + mass * mass / ( momentum * momentum ) );
+
+    return pathLength * centimeter * ( 1. / c_light ) * inverseBeta / nanosecond;
+}
+// *******************************************************************************************************
+
+
+
+// *******************************************************************************************************
+/*
+double 
+calculateBTofTstart( const StEvent* event, const double& bTofClockRange = 51200. )
+{
```

Comment:
Why this function is commented out? Even if it is not used, it can still be compiled and included in the library.

Reply:
I neither wrote, nor used or understood that piece of code well enough to tell if it is of any use at all. 
I deleted it and will fix it if need be.